src/main/java/hudson/tasks/junit/JUnitStep.java

```diff
@@ -0,0 +1,185 @@
+package hudson.tasks.junit;
+
+import hudson.*;
+import hudson.model.Result;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.remoting.VirtualChannel;
+import org.apache.tools.ant.DirectoryScanner;
+import org.apache.tools.ant.types.FileSet;
+import org.jenkinsci.plugins.workflow.steps.AbstractStepDescriptorImpl;
+import org.jenkinsci.plugins.workflow.steps.AbstractStepImpl;
+import org.jenkinsci.plugins.workflow.steps.AbstractSynchronousStepExecution;
+import org.jenkinsci.plugins.workflow.steps.StepContextParameter;
+import org.jenkinsci.remoting.RoleChecker;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.logging.Logger;
+
+/**
+ * Workflow step that archives JUnit Test results and returns a summary of counts
+ * to the caller
+ */
+public class JUnitStep extends AbstractStepImpl {
+
+    private static final Logger LOGGER = Logger.getLogger(JUnitStep.class.getName());
+
+    @Nullable
+    private String location;
+
+    private boolean keepLongStdio;
+
+    @DataBoundConstructor
+    public JUnitStep(@Nullable String location) {
+        this.location = location;
+    }
+
+    @Nullable
+    public String getLocation() {
+        return location;
+    }
+
+    public boolean isKeepLongStdio() {
+        return keepLongStdio;
+    }
+
+    @DataBoundSetter
+    public void setLocation(@Nullable String location) {
+        this.location = location;
+    }
```

Comment:
Delete this. If `location` is defined in a `@DataBoundConstructor`, it should be `final` and there should be no `@DataBoundSetter`.